### PR TITLE
Mkl main

### PR DESCRIPTION
**Pull request by mirkoklukas**

### CODE REVIEW

File: b3d/pose.py [hunk 1]
```diff
@@ -239,6 +265,12 @@ def concatenate_poses(pose_list):
             jnp.concatenate([pose.quat for pose in pose_list], axis=0),
         )
 
+    def concat(self, poses, axis=0):
```
Comment (georgematheos) [inline b3d/pose.py:268]: I don't see this used anywhere -- does this work?  Or does this code need to expand the dimension of `self.pos` and `self.quat` before we concatenate, here?

(Happy to merge and fix this if we use it and it errors later.)

File: b3d/pose_utils.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    """
+    # TODO: assert rq <= 1.0, "rq should be <= 1"
+    _, keys = keysplit(key, 1, 2)
+
+    xs = rx * uniform_samples_from_disc(keys[0], N, d=3)
+    qs = unit_disc_to_sphere(rq * uniform_samples_from_disc(keys[1], N, d=3))
+    return Pose(xs, qs)
+
+
+class UniformPose(genjax.ExactDensity, genjax.JAXGenerativeFunction):
```
Comment (georgematheos) [inline b3d/pose_utils.py:84]: TODO for us all - unify `UniformPose`, `VMF`, `GaussianVMFPose` with the b3d versions of these.

File: b3d/io/video_input.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
+from b3d.types import Array
```
Comment (nishadgothoskar) [inline b3d/io/video_input.py:1]: @mirkoklukas we have this already in https://github.com/probcomp/b3d/blob/2b0a264d7c845848ad1cf5c43e46ecb1fb834dde/b3d/utils.py#L491

Comment (nishadgothoskar) [reply b3d/io/video_input.py:1]: oh never mind. i see you remove that duplication

File: b3d/pose_utils.py (hunk 2 above)
Comment (georgematheos) [reply b3d/pose_utils.py]: fixed

**Review by nishadgothoskar — APPROVED** [review]

**Pull request merged**